test/test_common/network_utility.h

```diff
@@ -1,13 +1,21 @@
 #pragma once
 
+#include "envoy/network/address.h"
+
 namespace Network {
 namespace Test {
 
 /**
- * @returns an unused port, either be getting one from a port server, or by searching
- * through the ports for an unused port.
+ * Determines if the passed in address and port is available for binding. If the port is zero,
+ * the OS should pick an unused port for the supplied address (e.g. for the loopback address).
```

Comment:
Please add a warning about the racy nature of this (a port is picked but someone else might take it in the brief period before the caller binds).